configure.ac

```diff
@@ -960,12 +1016,12 @@ AS_IF([test x"$enable_shared" != "xno"],
       [sharedlib_cflags="-fPIC"
        AS_CASE([$cc_basename,$host],
            [*gcc*,powerpc-*-linux*],
-           [mksharedlib="$CC -shared -mbss-plt \$(LDFLAGS)"],
+           [mksharedlib='$(CC) -shared -mbss-plt'],
            [[*,i[3456]86-*]],
            # Disable DT_TEXTREL warnings on Linux and BSD i386
            # See https://github.com/ocaml/ocaml/issues/9800
-           [mksharedlib="$CC -shared \$(LDFLAGS) -Wl,-z,notext"],
```

Comment:
This changes the order, relative to that originally given in #10835 which needs to be checked, but I don't think that it was ever correct for hard-coded flags to appear _after `$(LDFLAGS)`